plugins/modules/cloud/scaleway/scaleway_volume.py

```diff
@@ -123,14 +131,20 @@ def core(module):
 
     volumeByName = None
     for volume in volumes_json['volumes']:
+        if volume['project'] == project and volume['name'] == name:
+            volumeByName = volume
+
         if volume['organization'] == organization and volume['name'] == name:
             volumeByName = volume
 
     if state in ('present',):
+        if project is None:
+            project = organization
```

Comment:
Shouldn't this assignment happen earlier, and only `project` be used in the above `for` loops?